### PR TITLE
Generator interface implementation

### PR DESCRIPTION
**Pull request by LOBsTerr**

### CODE REVIEW

File: src/Generator/Generator.php [hunk 1]
```diff
@@ -17,7 +17,7 @@
  *
  * @package Drupal\Console\Core\Generator
  */
-abstract class Generator
+abstract class Generator implements GeneratorInterface
```
Comment (jmolivas) [inline src/Generator/Generator.php]: @LOBsTerr  Thanks for the PR.

By merging this change, we are going to break custom generators at custom and contrib modules.

What chould be our choices?
- Make not an announcement about this change and make it mandatory for 2.0.
- Show a message when excuting DrupalConsole providing instruction about what to do.

Comment (LOBsTerr) [reply src/Generator/Generator.php]: @jmolivas I prefer get a feedback as soon as possible. Plus we give users a chance already right use it, but if you  prefer to keep it until 2.0 for me it is also ok

Comment (LOBsTerr) [reply src/Generator/Generator.php]: @jmolivas I changed my mind, it is better to give sometime for users. Let' make our users happy

Comment (jmolivas) [reply src/Generator/Generator.php]: I will merge this PR and rollback this line to
```
abstract class Generator
```
Then update the docs to suggest people write generator classes as:
```
class myGenerator extends Generator implements GeneratorInterface
```

**Comment by jmolivas** [comment]
@LOBsTerr Thanks for your contribution, your PR was merged. This will be included in the next release.

**Pull request merged**